zipline/finance/performance/period.py

```diff
@@ -252,15 +306,24 @@ def calculate_performance(self):
 
         payout = self._get_payout_total(pt.positions)
 
-        total_at_start = self.starting_cash + self.starting_value
-        self.ending_cash = self.starting_cash + self.period_cash_flow + payout
+        # total at beginning of subperiod
```

Comment:
Checked this for idempotency, and looks like it is retained. :+1:
(It is currently possible to call `calculate_performance` multiple times per a given dt.)